### ISSUE TITLE
Unit test failed 

### ISSUE THREAD

**Issue by daisuke-motoki** [pfnet-research/chainer-chemistry#191]
```
_________________________________________ test_get_molnet_bbbp_dataset_with_smiles _________________________________________

    @pytest.mark.slow
    def test_get_molnet_bbbp_dataset_with_smiles():
        # test default behavior
        pp = AtomicNumberPreprocessor()
        datasets = molnet.get_molnet_dataset('bbbp', preprocessor=pp,
                                             return_smiles=True)
    
        assert 'smiles' in datasets.keys()
        assert 'dataset' in datasets.keys()
        smileses = datasets['smiles']
        datasets = datasets['dataset']
        assert len(smileses) == 3
        assert len(datasets) == 3
    
        # Test each train, valid and test dataset
        for i, dataset in enumerate(datasets):
            # --- Test dataset is correctly obtained ---
            index = np.random.choice(len(dataset), None)
            atoms, label = dataset[index]
    
            assert atoms.ndim == 1  # (atom, )
            assert atoms.dtype == np.int32
            # (atom from, atom to) or (edge_type, atom from, atom to) assert label.ndim == 1
            assert label.shape[0] == 1
            assert label.dtype == np.int32
>           assert len(dataset) == expect_bbbp_lengths[i]
E           assert 1630 == 1631
E            +  where 1630 = len(<chainer_chemistry.datasets.numpy_tuple_dataset.NumpyTupleDataset object at 0x2b5c61c03d30>)

datasets_tests/molnet_tests/test_molnet.py:81: AssertionError
```

**Comment by delta2323** [comment]
Thank you for reporting the issue. Unfortunately, I could not reproduce the error.

```
%py.test tests/datasets_tests/molnet_tests/test_molnet.py::test_get_molnet_bbbp_dataset 
===================================================== test session starts =====================================================
platform linux -- Python 3.6.1, pytest-3.0.7, py-1.4.33, pluggy-0.4.0
rootdir: /home/delta/dev/chainer-chemistry, inifile:
collected 6 items 

tests/datasets_tests/molnet_tests/test_molnet.py .

================================================== 1 passed in 2.63 seconds ===================================================
```

Could you let me know your environment setting, such as the version of Chainer, Chainer Chemistry, NumPy?

**Comment by daisuke-motoki** [comment]
It's not "test_get_molnet_bbbp_dataset" but "test_get_molnet_bbbp_dataset_with_smiles".
If you could not reproduce the error even if you tried with the above function, you might need to clean cache of data.
I put the error again with my environment
```
================================================================================================================== test session starts ==================================================================================================================
platform linux -- Python 3.6.5, pytest-3.6.1, py-1.5.3, pluggy-0.6.0
rootdir: /mnt/vol02/dmotoki/work/chem/chainer-chemistry, inifile:
collected 1 item                                                                                                                                                                                                                                        

tests/datasets_tests/molnet_tests/test_molnet.py F                                                                                                                                                                                                [100%]

======================================================================================================================= FAILURES ========================================================================================================================
_______________________________________________________________________________________________________ test_get_molnet_bbbp_dataset_with_smiles ________________________________________________________________________________________________________

    @pytest.mark.slow
    def test_get_molnet_bbbp_dataset_with_smiles():
        # test default behavior
        pp = AtomicNumberPreprocessor()
        datasets = molnet.get_molnet_dataset('bbbp', preprocessor=pp,
                                             return_smiles=True)
    
        assert 'smiles' in datasets.keys()
        assert 'dataset' in datasets.keys()
        smileses = datasets['smiles']
        datasets = datasets['dataset']
        assert len(smileses) == 3
        assert len(datasets) == 3
    
        # Test each train, valid and test dataset
        for i, dataset in enumerate(datasets):
            # --- Test dataset is correctly obtained ---
            index = np.random.choice(len(dataset), None)
            atoms, label = dataset[index]
    
            assert atoms.ndim == 1  # (atom, )
            assert atoms.dtype == np.int32
            # (atom from, atom to) or (edge_type, atom from, atom to) assert label.ndim == 1
            assert label.shape[0] == 1
            assert label.dtype == np.int32
>           assert len(dataset) == expect_bbbp_lengths[i]
E           assert 1630 == 1631
E            +  where 1630 = len(<chainer_chemistry.datasets.numpy_tuple_dataset.NumpyTupleDataset object at 0x2ac77bfd1668>)

tests/datasets_tests/molnet_tests/test_molnet.py:81: AssertionError
----------------------------------------------------------------------------------------------------------------- Captured stderr call ------------------------------------------------------------------------------------------------------------------
  3%|▎         | 55/2050 [00:00<00:03, 548.42it/s][22:11:38] Explicit valence for atom # 1 N, 4, is greater than permitted
[22:11:38] Explicit valence for atom # 6 N, 4, is greater than permitted
 13%|█▎        | 272/2050 [00:00<00:01, 902.17it/s][22:11:39] Explicit valence for atom # 6 N, 4, is greater than permitted
 26%|██▌       | 531/2050 [00:00<00:01, 1057.60it/s][22:11:39] Explicit valence for atom # 11 N, 4, is greater than permitted
[22:11:39] Explicit valence for atom # 12 N, 4, is greater than permitted
[22:11:39] Explicit valence for atom # 5 N, 4, is greater than permitted
[22:11:39] Explicit valence for atom # 5 N, 4, is greater than permitted
[22:11:39] Explicit valence for atom # 5 N, 4, is greater than permitted
[22:11:39] Explicit valence for atom # 5 N, 4, is greater than permitted
[22:11:39] Explicit valence for atom # 5 N, 4, is greater than permitted
 32%|███▏      | 664/2050 [00:00<00:01, 1102.11it/s][22:11:39] Explicit valence for atom # 5 N, 4, is greater than permitted
 43%|████▎     | 891/2050 [00:00<00:01, 1070.09it/s][22:11:39] Conflicting single bond directions around double bond at index 1.
[22:11:39]   BondStereo set to STEREONONE and single bond directions set to NONE.
100%|██████████| 2050/2050 [00:01<00:00, 1055.27it/s]
------------------------------------------------------------------------------------------------------------------- Captured log call -------------------------------------------------------------------------------------------------------------------
csv_file_parser.py         122 WARNING  parse(), type: AssertionError, ()
=============================================================================================================== 1 failed in 6.52 seconds ================================================================================================================
```

**Comment by corochann** [comment]
Maybe related to RdKit version?

As you can see, there are 2050 targed to be parsed in total, but the number of success result == 1631 is tested in this case.
Preprocessing successes/fails for some smiles depending on the rdkit version.

**Comment by daisuke-motoki** [comment]
I see. Version 2018.03.02 of RDKit is used in my environment. Which version do you recommend?

**Comment by delta2323** [comment]
Sorry, I tested wrong a test. But `test_get_molnet_bbbp_dataset_with_smiles` was still successful in my environment (OS=macOS X, version `2017.09.3`). 

```
%(master)[]python -c "import rdkit; print(rdkit.__version__)"                                                 
2017.09.3
(anaconda3) /Users/oonokenta/dev/chainer-chemistry% [20:01:00]
%(master)[]py.test tests/datasets_tests/molnet_tests/test_molnet.py::test_get_molnet_bbbp_dataset_with_smiles 
===================================================== test session starts =====================================================
platform darwin -- Python 3.6.0, pytest-3.0.5, py-1.4.32, pluggy-0.4.0
rootdir: /Users/oonokenta/dev/chainer-chemistry, inifile: 
plugins: cov-2.3.1
collected 9 items 

tests/datasets_tests/molnet_tests/test_molnet.py .

================================================== 1 passed in 8.10 seconds ===================================================
```

But when I changed the version of RDKit to `2018.03.2`, the problem occurs.

**Comment by delta2323** [comment]
I notice that although the test failed, the detail message is different from @daisuke-motoki's one.

```
%(master)[]python -c "import rdkit; print(rdkit.__version__)"                                                 
2018.03.2
(rdkit-nightly) PHOBOS /home/delta/dev/chainer-chemistry% [20:11:53]
%(master)[]py.test tests/datasets_tests/molnet_tests/test_molnet.py::test_get_molnet_bbbp_dataset_with_smiles 
===================================================== test session starts =====================================================
platform linux -- Python 3.6.5, pytest-3.5.1, py-1.5.3, pluggy-0.6.0
rootdir: /home/delta/dev/chainer-chemistry, inifile:
plugins: remotedata-0.2.1, openfiles-0.3.0, doctestplus-0.1.3, arraydiff-0.2
collected 1 item                                                                                                              

tests/datasets_tests/molnet_tests/test_molnet.py F                                                                      [100%]

========================================================== FAILURES ===========================================================
__________________________________________ test_get_molnet_bbbp_dataset_with_smiles ___________________________________________

    @pytest.mark.slow
    def test_get_molnet_bbbp_dataset_with_smiles():
        # test default behavior
        pp = AtomicNumberPreprocessor()
        datasets = molnet.get_molnet_dataset('bbbp', preprocessor=pp,
                                             return_smiles=True)
    
        assert 'smiles' in datasets.keys()
        assert 'dataset' in datasets.keys()
        smileses = datasets['smiles']
        datasets = datasets['dataset']
        assert len(smileses) == 3
        assert len(datasets) == 3
    
        # Test each train, valid and test dataset
        for i, dataset in enumerate(datasets):
            # --- Test dataset is correctly obtained ---
            index = np.random.choice(len(dataset), None)
            atoms, label = dataset[index]
    
            assert atoms.ndim == 1  # (atom, )
            assert atoms.dtype == np.int32
            # (atom from, atom to) or (edge_type, atom from, atom to) assert label.ndim == 1
            assert label.shape[0] == 1
            assert label.dtype == np.int32
>           assert len(dataset) == expect_bbbp_lengths[i]
E           assert 1632 == 1631
E            +  where 1632 = len(<chainer_chemistry.datasets.numpy_tuple_dataset.NumpyTupleDataset object at 0x7fb8a30ed0f0>)

tests/datasets_tests/molnet_tests/test_molnet.py:81: AssertionError
---------------------------------------------------- Captured stderr call -----------------------------------------------------
  3%|▎         | 54/2050 [00:00<00:03, 535.04it/s][20:11:57] Explicit valence for atom # 1 N, 4, is greater than permitted
[20:11:57] Explicit valence for atom # 6 N, 4, is greater than permitted
 19%|█▉        | 391/2050 [00:00<00:01, 971.66it/s][20:11:57] Explicit valence for atom # 6 N, 4, is greater than permitted
 26%|██▌       | 538/2050 [00:00<00:01, 1070.36it/s][20:11:57] Explicit valence for atom # 11 N, 4, is greater than permitted
[20:11:57] Explicit valence for atom # 12 N, 4, is greater than permitted
[20:11:57] Explicit valence for atom # 5 N, 4, is greater than permitted
[20:11:57] Explicit valence for atom # 5 N, 4, is greater than permitted
[20:11:57] Explicit valence for atom # 5 N, 4, is greater than permitted
[20:11:57] Explicit valence for atom # 5 N, 4, is greater than permitted
[20:11:57] Explicit valence for atom # 5 N, 4, is greater than permitted
 33%|███▎      | 676/2050 [00:00<00:01, 1121.58it/s][20:11:57] Explicit valence for atom # 5 N, 4, is greater than permitted
 44%|████▍     | 909/2050 [00:00<00:01, 1091.79it/s][20:11:58] Conflicting single bond directions around double bond at index 1.
[20:11:58]   BondStereo set to STEREONONE and single bond directions set to NONE.
100%|██████████| 2050/2050 [00:01<00:00, 1235.77it/s]
------------------------------------------------------ Captured log call ------------------------------------------------------
data_frame_parser.py       129 WARNING  parse(), type: AssertionError, ()
================================================== 1 failed in 3.26 seconds ===================================================
```

**Comment by mottodora** [comment]
Can you change the `isomericSmiles` argument to False in `MolToSmiles` ?

**Comment by mottodora** [comment]
Maybe this line.
https://github.com/pfnet-research/chainer-chemistry/blob/a54b464dc2ec90b6dfce7326bfd981024cd91d8b/chainer_chemistry/dataset/preprocessors/mol_preprocessor.py#L29

**Comment by delta2323** [comment]
Unformtunately, the problem remains by the following change

```
%(master)[U]git diff
diff --git a/chainer_chemistry/dataset/preprocessors/mol_preprocessor.py b/chainer_chemistry/dataset/preprocessors/mol_preprocessor.py
index 891656a..a2b8594 100644
--- a/chainer_chemistry/dataset/preprocessors/mol_preprocessor.py
+++ b/chainer_chemistry/dataset/preprocessors/mol_preprocessor.py
@@ -26,7 +26,7 @@ class MolPreprocessor(BasePreprocessor):
         # we should re-obtain smiles from `mol`, so that the
         # smiles order does not contradict with input_features'
         # order.
-        smiles = Chem.MolToSmiles(mol)
+        smiles = Chem.MolToSmiles(mol, isometricSmiles=False)
         mol = Chem.MolFromSmiles(smiles)
         if self.add_Hs:
             mol = Chem.AddHs(mol)
```

**Comment by delta2323** [comment]
Good news. When we add the option here, too. The error disappears.
https://github.com/pfnet-research/chainer-chemistry/blob/a54b464dc2ec90b6dfce7326bfd981024cd91d8b/chainer_chemistry/dataset/preprocessors/mol_preprocessor.py#L33

**Comment by mottodora** [comment]
This problem is solved by #237 
When you encounter same problem, please re-open issue.
